lib/src/configuration.dart

```diff
@@ -48,11 +48,13 @@ typedef InitialDataCallback = void Function(Realm realm);
 /// Configuration used to create a [Realm] instance
 /// {@category Configuration}
 abstract class Configuration {
-
-  /// The default realm filename to be used. 
+  /// The default realm filename to be used.
   static String get defaultRealmName => _path.basename(defaultRealmPath);
   static set defaultRealmName(String name) => defaultRealmPath = _path.join(_path.dirname(defaultRealmPath), _path.basename(name));
-  
+
+  /// A collection of [SchemaObject] that will be used to construct the
+  /// [RealmSchema] once the Realm is opened.
+  final Iterable<SchemaObject> schemaObjects;
```

Comment:
Note for reviewers: this is replacing the `RealmSchema` field that previously existed on the configuration. The reason is that it's now possible to provide an empty collection and have the schema be read from the file. We're now exposing the `schema` on the `Realm` instance directly as that can only be truly known after we've opened the file. I wanted to make that distinction more explicit, which is why I opted to change the field on the configuration (otherwise we'd have `realm.schema != realm.config.schema` and that felt very awkward).